pkgs/by-name/ne/nextpnr/package.nix

```
@@ -47,8 +47,12 @@ stdenv.mkDerivation rec {
     ++ (lib.optional enableGui qtbase)
     ++ (lib.optional stdenv.cc.isClang llvmPackages.openmp);
 
-  cmakeFlags =
-    [ "-DCURRENT_GIT_VERSION=${lib.substring 0 7 (lib.elemAt srcs 0).rev}"
+  cmakeFlags = let
+    # the specified version must always start with "nextpnr-", so add it if
+    # missing (e.g. if the user overrides with a git hash)
+    rev = (lib.elemAt srcs 0).rev;
```

```suggestion
    rev = main_src.rev;
```
Is there a reason why this seems to be unnecessarily complicated? To be fair, `(lib.elemAt srcs 0).rev` was also used before but it's both more verbose and also harder to read IMO.

---

This was an artifact from before that name existed. Smart idea to fix, thanks.